iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/AbstractConnectable.java

```diff
@@ -175,22 +175,40 @@ protected void move(TerminalExt oldTerminal, TopologyPoint oldTopologyPoint, Str
                 .build();
 
         // detach the terminal from its previous voltage level
-        attachTerminal(oldTerminal, oldTopologyPoint, (VoltageLevelExt) bus.getVoltageLevel(), terminalExt);
+        replaceTerminal(oldTerminal, ((VoltageLevelExt) bus.getVoltageLevel()).getTopologyModel(), terminalExt, true);
     }
 
-    private void attachTerminal(TerminalExt oldTerminal, TopologyPoint oldTopologyPoint, VoltageLevelExt voltageLevel, TerminalExt terminalExt) {
+    void replaceTerminal(TerminalExt oldTerminal, TopologyPoint oldTopologyPoint, TerminalExt newTerminalExt, boolean notify) {
+        Objects.requireNonNull(oldTerminal);
+        int iSide = terminals.indexOf(oldTerminal);
+        if (iSide == -1) {
+            throw new PowsyblException("Terminal to replace not found");
+        }
+        terminals.set(iSide, newTerminalExt);
+
+        if (newTerminalExt != null && notify) {
```

Comment:
Then to be in line with the other replace we could remove this check also, at the cost of a Objects.requireNonNull

Reply:
Done